src/main/java/net/datafaker/transformations/XmlTransformer.java

```diff
@@ -0,0 +1,174 @@
+package net.datafaker.transformations;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringJoiner;
+
+import net.datafaker.sequence.FakeSequence;
+
+public class XmlTransformer<IN> implements Transformer<IN, CharSequence> {
+
+    private static final Map<Character, String> ESCAPING_MAP = createEscapeMap();
+    private static final int INDENTATION_STEP = 4;
+
+    private int tagIndex = 0;
+
+    private final boolean pretty;
+
+    private XmlTransformer(boolean pretty) {
+        this.pretty = pretty;
+    }
+
+    @Override
+    public CharSequence apply(IN input, Schema<IN, ?> schema) {
+        StringBuilder sb = new StringBuilder();
+        Arrays.stream(schema.getFields()).forEach(it -> apply(input, sb, it));
+        return sb.toString();
+    }
+
+    @Override
+    public String generate(FakeSequence<IN> input, Schema<IN, ?> schema) {
+        if (input.isInfinite()) {
+            throw new IllegalArgumentException("The sequence should be finite of size");
+        }
+
+        StringJoiner data = new StringJoiner(LINE_SEPARATOR);
+        for (IN in : input) {
+            data.add(apply(in, schema));
+        }
+
+        return data.toString();
+    }
+
+    @Override
+    public CharSequence generate(Schema<IN, ?> schema, int limit) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < limit; i++) {
+            sb.append(apply(null, schema));
+            if (i < limit - 1) {
+                sb.append(LINE_SEPARATOR);
+            }
+        }
+        return sb.toString();
+    }
+
+    public static class XmlTransformerBuilder<IN> {
+
+        private boolean pretty= false;
+
+        public XmlTransformer.XmlTransformerBuilder<IN> pretty(boolean pretty) {
+            this.pretty = pretty;
+            return this;
+        }
+
+        public XmlTransformer<IN> build() {
+            return new XmlTransformer<>(pretty);
+        }
+    }
+
+    private void apply(IN input, StringBuilder sb, Field<IN, ?> xmlNode) {
+
+        if (pretty && tagIndex > 0) {
+            sb.append(System.lineSeparator()).append(offset(tagIndex));
+        }
+
+        final String tag = xmlNode.getName().trim();
+        sb.append("<").append(tag);
+        if (xmlNode instanceof CompositeField) {
+            Field<IN, ?>[] attrs = ((CompositeField) xmlNode).getFields();
+            applyAttributes(input, sb, attrs);
+
+            Field<IN, ?> content = Arrays.stream(attrs)
+                .filter(inField -> !isAttribute(inField.getName())).findFirst()
+                .orElse(null);
+            applyTag(input, sb, content, tag);
+        } else {
+            applyTag(input, sb, xmlNode, tag);
+        }
```

Comment:
This could a bit simpler like
```suggestion
            xmlNode = Arrays.stream(attrs)
                .filter(inField -> !isAttribute(inField.getName())).findFirst()
                .orElse(null);
        }        
        applyTag(input, sb, xmlNode, tag);
```